File: portal/test_ingest.js
import fetch from "node-fetch";

const API_URL = "http://localhost:8000/ingest/ingest/sensor";

async function sendReading() {
  const payload = {
    device_id: "device-1",
    temperature: (20 + Math.random() * 10).toFixed(2),
    humidity: (30 + Math.random() * 30).toFixed(2),
    pm25: (5 + Math.random() * 45).toFixed(2),
    co2: (350 + Math.random() * 550).toFixed(2),
    health: (0.7 + Math.random() * 0.3).toFixed(2),
    risk_score: (0.1 + Math.random() * 0.8).toFixed(2),
    confidence: (0.5 + Math.random() * 0.5).toFixed(2),
    timestamp: new Date().toISOString()
  };

  console.log("Sending:", payload);

  const res = await fetch(API_URL, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload)
  });

  console.log("Status:", res.status);
  console.log("Response:", await res.text());
}

setInterval(sendReading, 2000);

File: portal/test_ingest.py
import requests
import time
import random

API_URL = "http://localhost:8000/ingest/ingest/sensor"

def send_reading():
    payload = {
        "device_id": "device-1",
        "temperature": round(random.uniform(20, 30), 2),
        "humidity": round(random.uniform(30, 60), 2),
        "pm25": round(random.uniform(5, 50), 2),
        "co2": round(random.uniform(350, 900), 2),
        "health": round(random.uniform(0.7, 1.0), 2),
        "risk_score": round(random.uniform(0.1, 0.9), 2),
        "confidence": round(random.uniform(0.5, 1.0), 2),
        "timestamp": time.strftime("%Y-%m-%dT%H:%M:%SZ", time.gmtime())
    }

    print("Sending:", payload)
    r = requests.post(API_URL, json=payload)
    print("Status:", r.status_code)
    print("Response:", r.text)

if __name__ == "__main__":
    while True:
        send_reading()
        time.sleep(2)